x-pack/test/functional/services/ml/job_wizard_geo.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export function MachineLearningJobWizardGeoProvider({ getService }: FtrProviderContext) {
+  const comboBox = getService('comboBox');
+  const testSubjects = getService('testSubjects');
+
+  return {
+    async assertGeoFieldInputExists() {
+      await testSubjects.existOrFail('mlGeoFieldNameSelect > comboBoxInput');
+    },
+
+    async assertGeoFieldSelection(expectedIdentifier: string[]) {
+      const comboBoxSelectedOptions = await comboBox.getComboBoxSelectedOptions(
+        'mlGeoFieldNameSelect > comboBoxInput'
+      );
+      expect(comboBoxSelectedOptions).to.eql(
+        expectedIdentifier,
+        `Expected geo field selection to be '${expectedIdentifier}' (got '${comboBoxSelectedOptions}')`
+      );
+    },
+
+    async selectGeoField(identifier: string) {
+      await comboBox.set('mlGeoFieldNameSelect > comboBoxInput', identifier);
+      await this.assertGeoFieldSelection([identifier]);
+    },
+
+    async assertSplitFieldInputExists() {
```

Comment:
This service method and many other in this file already exist in the `job_wizard_multi_metric` service. It looks like we share some wizard components here, so I'd like to avoid duplicate service methods. I think we have two feasible options here: we could either move these methods to the `job_wizard_common` service or we make the geo wizard test call methods from the multi metric wizard service.

Reply:
Fixed in 334b359ed606846e8069b1e70bd2e997a39bd5b3